### ISSUE TITLE
Coin supply discrepancy

### ISSUE THREAD

**Issue by xaur** [decred/dcrdata#1963]
Home page shows 14.9M, charts show 15.2M. There might be a discrepancy in the API too. Total supply affects percentages based on it (staked %, mixed %) and the numbers/chart we report in DJ.

https://dcrdata.decred.org/ v6.2.0-pre+dev

![home_supply](https://github.com/decred/dcrdata/assets/24484727/fd521426-14ed-472e-b32b-c9465454fe4b)

![chart_circ_supply](https://github.com/decred/dcrdata/assets/24484727/347aa9a0-7d94-4312-a3cc-f5cc202a5bb6)

![chart_stakepart_supply](https://github.com/decred/dcrdata/assets/24484727/9165c40c-e564-4870-9ee5-9abf93fdaf23)

**Comment by chappjc** [comment]
Unfortunately the charts data code has no reorg handling.  It's all append-only.
I'm not too clear on the operation of the code in db/cache/charts.go, but it's going to need some work.

**Comment by chappjc** [comment]
Correction, it does have reorg handling.  I'm not sure why the charts are wrong.

**Comment by xaur** [comment]
Well at least we know which one is wrong now. We'll be using supply on the [home page](https://dcrdata.decred.org/) (14.9M as of writing) to calculate percentages of staked/mixed/etc.

**Comment by xaur** [comment]
I'm thinking if we can (or should) work-around and correct the staked/mixed percentages (on charts like [this](https://github.com/bochinchero/dcrsnapshots/blob/main/2023-04/700/Staking_Daily_Participation_PC.png)) so it's worth checking the other inputs too.

1. Do historical "staked DCR" values look correct? It [shows](https://dcrdata.decred.org/charts?chart=stake-participation&zoom=l3eju0ni-lhmph4ov&bin=day&axis=time&visibility=true-true-true) `9,622,254` for 2023-05-14 (yesterday) with `bin=day`.
2. Do historical "mixed DCR" values look correct? It [shows](https://dcrdata.decred.org/charts?chart=coin-supply&zoom=l3wpb0e7-lj3mddi3&bin=day&axis=time&visibility=true-true-true) `9,404,889` for 2023-05-14 (yesterday) with `bin=day`.
3. Is there another API that returns correct historical DCR circulating supply values?

**Comment by chappjc** [comment]
The magnitude of the coin supply discrepancy appears to be close to the total treasurybase amount since those began about 211666 blocks ago.  Looking further.

**Comment by chappjc** [comment]
@davecgh https://github.com/decred/dcrd/blob/c6e53c9ef0926f172a637a6552b744cbd6445537/internal/blockchain/subsidy.go#L183-L188 Is treasurybase caught by this?  It's the first tx in the stake tree, I believe, and I would not expect `stake.IsSSGen` to match it.

**Comment by davecgh** [comment]
No, it's not, so that is indeed incorrect.  It's only used in reporting the total coin supply from the RPC server, so it's not any concern from a consensus standpoint.  I'll add it to my list to fix.  Just adding the treasurybase there will fix it moving forward, but since the best state is stored in the database, it's going to need a one-time recalculation.

**Comment by davecgh** [comment]
The underlying RPC issue is resolved by https://github.com/decred/dcrd/issues/3112.

**Comment by davecgh** [comment]
Upstream merged, so I believe all dcrdata should need to do is update to the latest master.
